src/m365/spo/commands/site/site-admin-add.ts

```diff
@@ -0,0 +1,358 @@
+import { Logger } from '../../../../cli/Logger.js';
+import config from '../../../../config.js';
+import GlobalOptions from '../../../../GlobalOptions.js';
+import request, { CliRequestOptions } from '../../../../request.js';
+import { entraGroup } from '../../../../utils/entraGroup.js';
+import { formatting } from '../../../../utils/formatting.js';
+import { FormDigestInfo, spo } from '../../../../utils/spo.js';
+import { validation } from '../../../../utils/validation.js';
+import SpoCommand from '../../../base/SpoCommand.js';
+import commands from '../../commands.js';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface AdminUserResult {
+  email: string;
+  loginName: string;
+  name: string;
+  userPrincipalName: string;
+}
+
+interface AdminResult {
+  value: AdminUserResult[];
+}
+
+interface IGraphUser {
+  userPrincipalName: string;
+}
+
+interface ISiteOwner {
+  LoginName: string;
+}
+
+interface ISPSite {
+  Id: string;
+}
+
+interface ISiteUser {
+  Id: number;
+}
```

Comment:
We will probably use these in different commands. Wouldn't it make more sense to create a separate model file for this?

Reply:
ok i added it to SiteAdmin.ts file, It is also used in site admin list command